python/rtypes/dataframe/trigger_manager.py

```diff
@@ -156,12 +157,47 @@ def __execute_trigger(self, tp, time, action, dataframe, new, old, current):
 
             Returns:
                 None: Does not return anything, only activates procedure objects
-        """
-        try:
-            for procedure in self.__get_trigger(tp, time, action):
+            __rtypes_current_triggers__ = {"before_update": [Customer, Transaction]}
+                """
+        for procedure in self.__get_trigger(tp, time, action):
+            """
+            The process below is used to prevent update triggers from creating infinite
+            recursive loops.
+
+            There is a map that is used to do this.
+
+            The map's keys are each procedure object, each key has a set of objets.
+
+            Each set is a literal set of object.
+
+            These objects show that the procedure has activated on the objects inside
+                the set.
+
+            If there isn't an object in the set, then the trigger hasn't neen activated
+                on it and is free to do so
+            """
+            # update
+            if action == TriggerAction.update and (new or old or current):
+
+                # if the triggers has been mapped to an object
+                if procedure in self.__rtypes_current_triggers__:
+
+                    # If the object is in this map, then dont do anything
+                    if new in self.__rtypes_current_triggers__[procedure]:
+                        pass
```

Comment:
You should put return here instead of pass. If we add code later after the if-else statements, it is going to break.